src/main/java/core/basesyntax/ArrayList.java

```
@@ -1,49 +1,98 @@
     public void add(T value) {
-
+        defaultFull();
+        defaultArray[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > size || index < 0) {
+            throw new ArrayIndexOutOfBoundsException("Can't add element with this index");
+        }
+        defaultFull();
+        System.arraycopy(defaultArray, index, defaultArray, index + 1, size - index);
+        defaultArray[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index >= size || index < 0) {
+            throw new ArrayIndexOutOfBoundsException("Can't get element with this index");
+        }
+        return defaultArray[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        if (index >= size || index < 0) {
+            throw new ArrayIndexOutOfBoundsException("Can't set element with this index");
+        }
+        defaultArray[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        if (index >= size || index < 0) {
+            throw new ArrayIndexOutOfBoundsException("Can't remove element with this index");
+        }
+        T removeElement = defaultArray[index];
+        System.arraycopy(defaultArray, index + 1, defaultArray, index, size - index - 1);
+        defaultArray [--size] = null;
+        return removeElement;
     }
 
     @Override
     public T remove(T t) {
-        return null;
+        for (int i = 0; i < size; i++) {
+            if ((t == defaultArray[i]) || (t != null) && (t.equals(defaultArray[i]))) {
+                remove(i);
+                break;
+            } else if (i == size - 1) {
+                throw new NoSuchElementException("This element don't exist in this collection");
```

we can throw this exception instead of return on line 78

---

Made